frame/staking-rewards/src/lib.rs

```diff
@@ -228,20 +252,21 @@ pub mod pallet {
 	>;
 
 	#[pallet::type_value]
-	pub fn SharesOnEmpty<T: Config>() -> u128 {
-		u128::zero()
+	pub fn SharesOnEmpty<T: Config>() -> ShareAmount {
+		ShareAmount::zero()
 	}
 
+	//// active running total shares
 	#[pallet::storage]
-	#[pallet::getter(fn total_shares)]
-	pub type TotalShares<T: Config> = StorageMap<
-		_,
-		Twox64Concat,
-		(AssetIdOf<T>, AssetIdOf<T>),
-		u128,
-		ValueQuery,
-		SharesOnEmpty<T>,
-	>;
+	#[pallet::getter(fn running_total_shares)]
+	pub type RunningTotalShares<T: Config> =
+		StorageMap<_, Twox64Concat, AssetIdOf<T>, ShareAmount, ValueQuery, SharesOnEmpty<T>>;
+
+	/// pending total shares to be applied on next run
```

Comment:
Would be nice to have better docs on why this storage exists

Reply:
total stake is depend on this. you should not modify total stake until reward happend.